plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -155,6 +155,17 @@ public boolean hasField(String path) {
      * @throws IllegalArgumentException if the path is null, empty, invalid or if the field doesn't exist.
      */
     public void removeField(String path) {
+        removeField(TemplateService.wrap(path));
```

Comment:
I think that depends on if all or almost all processors end up using templates. Right now in this PR only remove and set processors use templates, so I kept the old method signatures around to not make the change bigger than it is.